src/zeek.bif

```diff
@@ -5015,6 +5015,22 @@ function generate_all_events%(%) : bool
 	return zeek::val_mgr->True();
 	%}
 
+## Check if an event is handled. Typically this means that a script defines an event.
+## This currently is mainly used to warn when events are defined that will not be used
+## in certain conditions.
+##
+## event_name: event name to check
+##
+## returns: true if the named event is handled.
+function is_event_handled%(event_name: string%) : bool
```

Comment:
Have you considered using `ev: any` instead of a string to prevent some potential string typoing by being able to use the event identifier directly (at the expense of needing dynamic type checking within the BiF).
```
if ( ! is_event_handled(dns_EDNS_addl) ) { }
```

In any case, with `string` or `any`, this should call `builtin_error` if name or identifier isn't an event to notice usage errors (or typos). For probing if an event exists, once can already use `@ifdef ( dns_EDNS_addl )`.

Reply:
I am happy to add the builtin error.

nope, I have not considered making is possible to use this with an event handler directly. Unless you have strong feelings about this, I also don't see a super good reason to do this once the builtin error is added.

Reply:
No overly strong opinion. Resolving. Can add identifier support down the road if wanted without breaking existing users, too.